src/01_getting_started/02_why_async.md

```diff
@@ -1,45 +1,121 @@
+- **The actor model** divides all concurrent computation into units called
+  actors, which communicate through fallible message passing, much like
+  in distributed systems. It can be efficiently implemented, but it leaves
+  many practical issues unanswered, such as flow control and retry logic.
+
+In short, asynchronous programming allows highly performant implementations
+that are suitable for low-level languages like Rust, while providing
+most of the ergonomic benefits of threads and coroutines.
+
+## Async in Rust vs other languages
+
+Although asynchronous programming is supported in many languages, some
+details vary across implementations. Rust's implementation of async
+differs from most languages in a few ways:
+
+- **Futures are inert** in Rust and make progress only when polled. Dropping a
+  future stops it from making further progress.
+- **Async is zero-cost** in Rust, and does not require heap allocations or
+  dynamic dispatch.
+  As such, you can use async Rust in constrained environments, such as embedded
+  systems.
+- **No built-in runtime** is provided by Rust. Instead, runtimes are provided by
+  community maintained crates.
+- **Both single- and multithreaded** runtimes are available in Rust, which have
+  different strengths and weaknesses.
+
+## Async vs threads in Rust
+
+The most obvious alternative to async in Rust is using OS threads, either
+directly through [`std::thread`](https://doc.rust-lang.org/std/thread/)
+or indirectly through a thread pool.
+Migrating from threads to async or vice versa
+typically requires major refactoring work, both in terms of implementation and
+(if you are building a library) any exposed public interfaces. As such,
+picking the model that suits your needs early can save you a lot of time.
+
+Note that asynchronous programming is not _better_ than threads, but different.
+If you don't need async for performance reasons, threads can often be
+a simpler option. Specifically:
+
+**OS threads** are suitable for a small number of tasks, since threads come with
+CPU and memory overhead. Spawning and switching between threads
+is quite expensive—even idle threads consume valuable system resources.
+A thread pool library can help mitigate some of these costs, but not all.
+However, threads let you reuse existing synchronous code without significant
+code changes—no particular programming model is required.
+In some operating systems, you can also change the priority of a thread,
+which is useful for drivers and other latency sensitive applications.
+
+**Async** provides significantly reduced CPU and memory
+overhead, especially for workloads with a
+large amount of IO-bound tasks, such as servers and databases.
+All else equal, you can have orders of magnitude more tasks than OS threads.
+That said, async Rust can result in larger binary sizes, both due to the state
+machines that are generated from async functions and because each executable
+needs to bundle an async runtime.
+
```

Comment:
```suggestion
However, async Rust results in larger binary blobs due to the state
machines generated from async functions and since each executable
bundles an async runtime.

```